pkg/bootstrap/ssh.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2017 Kinvolk GmbH
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	privKeyPath string = "id_rsa"
+	pubKeyPath  string = "id_rsa.pub"
+)
+
+func generateSSHKeys() error {
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	privKeyFile, err := os.Create(privKeyPath)
+	if err != nil {
+		return err
+	}
+	defer privKeyFile.Close()
+
+	privKeyPem := &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privKey)}
+	if err := pem.Encode(privKeyFile, privKeyPem); err != nil {
+		return err
+	}
+
+	pubKey, err := ssh.NewPublicKey(&privKey.PublicKey)
+	if err != nil {
+		return err
+	}
+
+	if err := ioutil.WriteFile(pubKeyPath, ssh.MarshalAuthorizedKey(pubKey), 0655); err != nil {
```

Comment:
Not really, but the same thing happens when you generate keys with `ssh-keygen` - only the private stuff gets 0600 mode, while the pub key permissions are more open.

But no problem I can change it, probably no difference for us.